### PR TITLE
Store TW reaction ID and use it to mark rooms as read when necessary

### PR DESCRIPTION
**Pull request by smweber**

### CODE REVIEW

File: mautrix_twitter/portal.py [hunk 1]
```diff
@@ -504,7 +506,12 @@ async def _reupload_twitter_media(
...
-        self, sender: p.Puppet, msg_id: int, reaction: ReactionKey, time: datetime
+        self,
+        sender: p.Puppet,
+        msg_id: int,
+        reaction: ReactionKey,
+        time: datetime,
+        reaction_id: int,
     ) -> None:
         async with self._reaction_lock:
             dedup_id = (msg_id, sender.twid, reaction)
```
Comment (tulir) [inline mautrix_twitter/portal.py:517]:
```suggestion
            # TODO update the database with the reaction_id of outgoing reactions
            dedup_id = (msg_id, sender.twid, reaction)
```

Probably doesn't need to be implemented now, so let's just add a todo

File: mautrix_twitter/portal.py [hunk 2]
```diff
@@ -178,6 +179,7 @@ async def _upsert_reaction(
                 tw_receiver=self.receiver,
                 tw_sender=sender.twid,
                 reaction=reaction,
+                tw_reaction_id=reaction_id,
             ).insert()
```
Comment (tulir) [inline mautrix_twitter/portal.py:183]: The existing/edit case also needs to update the reaction ID

Comment (smweber) [reply mautrix_twitter/portal.py:185]: 👍

File: mautrix_twitter/db/reaction.py [hunk 3]
```diff
@@ -51,12 +52,15 @@ async def insert(self) -> None:
...
         )
 
-    async def edit(self, mx_room: RoomID, mxid: EventID, reaction: ReactionKey) -> None:
+    async def edit(
+        self, mx_room: RoomID, mxid: EventID, reaction: ReactionKey, tw_reaction_id: int | None
+    ) -> None:
         q = (
             "UPDATE reaction SET mxid=$1, mx_room=$2, reaction=$3 "
-            "WHERE tw_msgid=$4 AND tw_receiver=$5 AND tw_sender=$6"
+            "WHERE tw_msgid=$4 AND tw_receiver=$5 AND tw_sender=$6 AND tw_reaction_id=$7"
```
Comment (tulir) [inline mautrix_twitter/db/reaction.py:63]: This looks like it's the wrong way around (it should store the new reaction ID)

**Review by tulir — APPROVED** [review]
Otherwise looks fine

File: mautrix_twitter/db/reaction.py (hunk 3 above)
Comment (smweber) [reply mautrix_twitter/db/reaction.py:63]: You're totally right - even my editor was showing me I had the wrong one ;)

**Pull request merged**